src/manager/node.c

```diff
@@ -570,61 +569,36 @@ static int node_method_start_unit(sd_bus_message *m, void *userdata, UNUSED sd_b
-        Node *node = req->node;
-        Manager *manager = node->manager;
-        JobSetup *setup = req->userdata;
-
-        if (sd_bus_message_is_method_error(m, NULL)) {
-                /* Forward error */
-                return sd_bus_reply_method_error(setup->request_message, sd_bus_message_get_error(m));
-        }
-
-        if (!manager_add_job(manager, setup->job)) {
-                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
-        }
-
-        return sd_bus_reply_method_return(setup->request_message, "o", setup->job->object_path);
+static int node_method_start_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        return node_run_unit_lifecycle_method(m, (Node *) userdata, "Start", "StartUnit");
 }
 
-static int node_method_stop_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
-        Node *node = userdata;
-        _cleanup_agent_request_ AgentRequest *req = NULL;
-        const char *unit = NULL;
-        const char *mode = NULL;
+/*************************************************************************
+ ********** org.containers.hirte.Node.StopUnit ***************************
+ ************************************************************************/
 
-        int r = sd_bus_message_read(m, "ss", &unit, &mode);
-        if (r < 0) {
-                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_INVALID_ARGS, "Invalid arguments");
-        }
 
-        _cleanup_job_setup_ JobSetup *setup = job_setup_new(m, node, unit, "Stop");
-        if (setup == NULL) {
-                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Out of memory");
-        }
+static int node_method_stop_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        return node_run_unit_lifecycle_method(m, (Node *) userdata, "Stop", "StopUnit");
+}
 
-        req = node_create_request(
-                        node,
-                        "StopUnit",
-                        method_stop_unit_callback,
-                        job_setup_ref(setup),
-                        (free_func_t) job_setup_unref);
-        if (req == NULL) {
-                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
-        }
+/*************************************************************************
+ ********** org.containers.hirte.Node.RestartUnit ************************
+ ************************************************************************/
 
-        r = sd_bus_message_append(req->message, "ssu", unit, mode, setup->job->id);
-        if (r < 0) {
-                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
-        }
+static int node_method_restart_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        return node_run_unit_lifecycle_method(m, (Node *) userdata, "Restart", "RestartUnit");
```

Comment:
Actually, i looked at what systemd does and the "mode" is:
> JobType refers to the job's type and is one of start, verify-active, stop, reload, restart, try-restart, reload-or-start. Note that later versions might define additional values. 
So, no capital letters in the beginning.